src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,236 @@
+import React, {
+  CSSProperties, forwardRef, RefObject,
+} from 'react';
+import { DialogOverlay, DialogContent } from '@palmetto/dialog';
+import classNames from 'classnames';
+import { DimensionSize } from '../../types';
+import { WIDTH_OPTIONS } from '../../lib/tokens';
+import { Box } from '../Box/Box';
+import { Icon } from '../Icon/Icon';
+import styles from './Drawer.module.scss';
+
+export type DrawerPlacementType = 'left' | 'right' | 'top' | 'bottom';
+
+export const DRAWER_PLACEMENT: DrawerPlacementType[] = ['right', 'left', 'top', 'bottom'];
+
+export interface DrawerProps {
+  /**
+   * Handle zoom/pinch gestures on iOS devices when scroll locking is enabled.
+   */
+  allowPinchZoom?: boolean;
+  /**
+   * A drawer needs to be properly labeled to provide context for users with
+   * assistive technology such as screen readers. If a drawer is announced to
+   * the user without a label, it can be confusing and difficult to navigate.
+   */
+  ariaLabel?: string;
+  /**
+   * The id of the element that should be used as the drawer's label by assistive technologies like screen readers.
+   */
+  ariaLabelledBy?: string;
+  /**
+   * Contents of the dialog.
+   */
+  children?: React.ReactNode;
+  /**
+   * Additional class names to add to the drawer content.
+   */
+  className?: string;
+  /**
+   * Whether the drawer has a visible close button.
+   * If a title is defined, then a close button will be rendered
+   */
+  closeButton?: boolean;
+  /**
+   * If true, the drawer will close when the overlay is clicked
+   */
+  closeOnOverlayClick?: boolean;
```

Comment:
By "overlay" are you referring to the drawer itself or the area outside the drawer? Should this be something like `closeOnClickOutside`?